main.tf

```diff
@@ -1,11 +1,20 @@
 locals {
+  tenant_id = length(var.tenant_id) > 0 ? var.tenant_id : data.azurerm_subscription.primary.tenant_id
   application_id       = var.use_existing_ad_application ? var.application_id : module.az_cfg_ad_application.application_id
   application_password = var.use_existing_ad_application ? var.application_password : module.az_cfg_ad_application.application_password
+  subscription_ids = var.all_subscriptions ? (
+    // the user wants to grant access to all subscriptions
+    [for s in data.azurerm_subscriptions.available.subscriptions : s.subscription_id]
+    ) : (
+    // or, if the user wants to grant a list of subscriptions, if none then we default to the primary subscription
+    length(var.subscription_ids) > 0 ? var.subscription_ids : [data.azurerm_subscription.primary.subscription_id]
+  )
+  service_principal_id = var.use_existing_ad_application ? var.service_principal_id : module.az_cfg_ad_application.service_principal_id
```

Comment:
since we created this local variable, we need to update our outputs https://github.com/lacework/terraform-azure-config/blob/main/output.tf#L12-L15